adapters/handlers/grpc/v1/parse_search_request.go

```diff
@@ -1252,21 +1313,60 @@ OUTER:
 	}
 }
 
-func extractVector(vectors []*pb.Vectors) (models.Vector, error) {
-	if len(vectors) > 0 {
-		vec := vectors[0]
-		switch vec.Type {
+func extractVectors(vectors []*pb.Vectors) (interface{}, error) {
+	var vecs [][]float32
+	var multiVecs [][][]float32
+	for i := range vectors {
+		vec, err := extractVector(vectors[i])
+		if err != nil {
+			return nil, fmt.Errorf("vectors[%d]: %w", i, err)
+		}
+		switch v := vec.(type) {
+		case []float32:
+			vecs = append(vecs, v)
+		case [][]float32:
+			multiVecs = append(multiVecs, v)
+		default:
+			// do nothing
+		}
+	}
+	if len(multiVecs) > 0 {
+		return multiVecs, nil
+	}
+	return vecs, nil
+}
+
+func extractVector(vector *pb.Vectors) (models.Vector, error) {
```

Comment:
The same point here as above with `extractVectors`. I see that `models.Vector` is just an `interface{}` behind the scenes, do you think it's worth it to do the above here also?